### PR TITLE
Removed deprecated package Microsoft.CodeAnalysis.FxCopAnalyzers

### PR DESCRIPTION
**Pull request by michaelmajgaard**
## Description
Removed deprecated package Microsoft.CodeAnalysis.FxCopAnalyzers. Update package StyleCop.Analyzers. Fixed minor issues and cleaned up code.

## References
* #120 #116

### CODE REVIEW

File: source/Energinet.DataHub.PostOffice.Infrastructure/MessageReplyStorage/MessageReplyDictionaryStorage.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace Energinet.DataHub.PostOffice.Infrastructure.MessageReplyStorage
 {
     public class MessageReplyDictionaryStorage : IMessageReplyStorage
     {
-        private static readonly Dictionary<string, string> _savedReplyResponses = new ();
```
Comment (FirestarJes) [inline source/Energinet.DataHub.PostOffice.Infrastructure/MessageReplyStorage/MessageReplyDictionaryStorage.cs:24]: Rider gives a warning if you omit the space after new. But can't see if that will go away after merging, so i'm leaving this just as a comment, but it shouldn't prevent merging

**Review by FirestarJes — APPROVED** [review]

File: Directory.Build.props [hunk 2]
```diff
@@ -5,9 +5,8 @@
     -->
     <ItemGroup>
         <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.8.55" />
-        <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" />
+        <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.354" />
         <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2" />
-        <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.1" />
```
Comment (cmgrd) [inline Directory.Build.props:10]: Remember to add `<AnalysisMode>AllEnabledByDefault</AnalysisMode>`. As the built-in analyzers is much more conservative compared to FxCopAnalyzers. See [this](https://docs.microsoft.com/en-us/visualstudio/code-quality/migrate-from-fxcop-analyzers-to-net-analyzers?view=vs-2019#migration-steps) for a more in-depth explanation.

**Review by cmgrd — COMMENTED** [review]
Added a comment about the changes to the rules behavior.

File: Directory.Build.props (hunk 2 above)
Comment (michaelmajgaard) [reply Directory.Build.props:10]: Fixed :)

**Review by FirestarJes — APPROVED** [review]

File: Directory.Build.props (hunk 2 above)
Comment (cmgrd) [reply Directory.Build.props:10]: 👍

**Pull request merged**